test/e2e/router_compression_test.go

```diff
@@ -0,0 +1,226 @@
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestRouterCompression(t *testing.T) {
+	// Get the default ingress controller
+	ic, err := getIngressController(t, kclient, defaultName, 1*time.Minute)
+	if err != nil {
+		t.Fatalf("failed to get ingress controller: %v", err)
+	}
+
+	// Configure the default ingress controller to apply compression to the canary route's
+	// MIME type, which is "text/plain; charset=utf-8"
+	mimeType := []operatorv1.CompressionMIMEType{"text/plain; charset=utf-8"}
+	ic.Spec.HTTPCompression = operatorv1.HTTPCompressionPolicy{MimeTypes: mimeType}
+	if err := wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
+		if err := kclient.Update(context.TODO(), ic); err != nil {
+			t.Logf("failed to update ingress controller: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("failed to update ingress controller: %v", err)
+	}
+
+	// Cleanup the ic.Spec.CompressionPolicy
+	defer func() {
+		ic, err := getIngressController(t, kclient, defaultName, 1*time.Minute)
+		if err != nil {
+			t.Fatalf("failed to cleanup ingress controller: %v", err)
+		}
+		mimeType := []operatorv1.CompressionMIMEType{}
+		ic.Spec.HTTPCompression = operatorv1.HTTPCompressionPolicy{MimeTypes: mimeType}
+		// Remove the mimeType that was added
+		if err := wait.PollImmediate(10*time.Second, 1*time.Minute, func() (bool, error) {
+			if err := kclient.Update(context.TODO(), ic); err != nil {
+				t.Logf("failed to cleanup ingress controller: %v, retrying...", err)
+				return false, nil
+			}
+			return true, nil
+		}); err != nil {
+			t.Fatalf("failed to cleanup ingress controller: %v", err)
+		}
+	}()
+
+	// Test some other compression policies for the ingress config
+	mimeTypesNormative := []operatorv1.CompressionMIMEType{"text/html", "application/json", "x-custom/allow-custom"}
+	compressionPolicyNormative := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNormative}
+
+	mimeTypesEmpty := []operatorv1.CompressionMIMEType{}
+	compressionPolicyEmpty := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesEmpty}
+
+	mimeTypesNeedQuotes := []operatorv1.CompressionMIMEType{`text/html; v="keepquoted"`, `x-custom/allow-custom; specialChar='`}
+	compressionPolicyNeedQuotes := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesNeedQuotes}
+
+	mimeTypesErrors := []operatorv1.CompressionMIMEType{"text/", "x- /value", "//"}
+	compressionPolicyErrors := operatorv1.HTTPCompressionPolicy{MimeTypes: mimeTypesErrors}
+
+	icName := types.NamespacedName{Namespace: operatorNamespace, Name: "http-compression-1"}
+	domain := icName.Name + "." + dnsConfig.Spec.BaseDomain
+	if err := testCompressionPolicy(t, icName, domain, compressionPolicyNormative); err != nil {
+		log.Fatalf("error testing normal compression policy: %v", err)
```

Comment:
Tests usually don't use `log`.  Why not use `t` here and below?  
```suggestion
		t.Errorf("error testing normal compression policy: %v", err)
```

Reply:
Thanks.